components/cosmetic_filters/browser/cosmetic_filters_resources.cc

```diff
@@ -0,0 +1,123 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/cosmetic_filters/browser/cosmetic_filters_resources.h"
+
+#include <utility>
+
+#include "base/json/json_reader.h"
+#include "base/optional.h"
+#include "base/values.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+
+namespace cosmetic_filters {
+
+namespace {
+
+base::Optional<base::Value> HiddenClassIdSelectorsOnTaskRunner(
+    brave_shields::AdBlockService* ad_block_service,
+    const std::vector<std::string>& classes,
+    const std::vector<std::string>& ids,
+    const std::vector<std::string>& exceptions) {
+  return ad_block_service->HiddenClassIdSelectors(classes, ids, exceptions);
+}
+
+base::Optional<base::Value> UrlCosmeticResourcesOnTaskRunner(
+    brave_shields::AdBlockService* ad_block_service,
+    const std::string& url) {
+  return ad_block_service->UrlCosmeticResources(url);
+}
+
+}  // namespace
+
+CosmeticFiltersResources::CosmeticFiltersResources(
+    HostContentSettingsMap* settings_map,
+    brave_shields::AdBlockService* ad_block_service)
+    : settings_map_(settings_map), ad_block_service_(ad_block_service) {}
+
+CosmeticFiltersResources::~CosmeticFiltersResources() {}
+
+void CosmeticFiltersResources::HiddenClassIdSelectors(
+    const std::string& input,
+    const std::vector<std::string>& exceptions,
+    HiddenClassIdSelectorsCallback callback) {
+  base::Optional<base::Value> input_value = base::JSONReader::Read(input);
```

Comment:
I'd rather do all this stuff in the function that works on task runner, not a blocker though